multitenant: split at tenant's end key range boundary

ecwall opened the pull request:
Fixes #92072

Release note: None

aliher1911 committed:
loqrecovery: add replica info collection to admin server

This commit adds replica info and range descriptor collection from
partially available cluster that lost quorum on some ranges.
Collection is done using AdminServer calls. Cluster wide call
performs fan-out using node info from gossip. Local replica info
collection on nodes is done by scanning storages.

Release note (cli change): File format used for transient loss of
quorum recovery files has changed. It is not possible to use replica
info files generated by earlier versions to be used with current and
future versions.

ecwall committed:
multitenant: support crdb_internal.list_sql_keys_in_range() for secondary tenants

Fixes #95006

The crdb_internal.list_sql_keys_in_range builtin now uses the RangeDescIterator
and scopes the range span by the tenant span to prevent these errors:
```
ERROR: RangeIterator failed to seek to /Meta2/"\x00": rpc error: code = Unauthenticated desc = requested key /Meta2/"\x00" not fully contained in tenant keyspace /Tenant/1{2-3}
```

Release note: None

ecwall committed:
multitenant: split at tenant's end key range boundary

Fixes #92072

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95216)
<!-- Reviewable:end -->

knz committed:
sql: populate `pg_proc.prokind`

Release note (bug fix): The `prokind` column of `pg_catalog.pg_proc`
is now populated properly.
